app/bundles/CoreBundle/Model/FormModel.php

```diff
@@ -50,10 +50,24 @@ public function isLocked($entity)
     {
         if (method_exists($entity, 'getCheckedOut')) {
             $checkedOut = $entity->getCheckedOut();
-            if (!empty($checkedOut)) {
-                //is it checked out by the current user?
+            if (!empty($checkedOut) && $checkedOut instanceof \DateTime) {
                 $checkedOutBy = $entity->getCheckedOutBy();
-                if (!empty($checkedOutBy) && $checkedOutBy !== $this->userHelper->getUser()->getId()) {
+                $maxLockTime  = $this->factory->getParameter('max_entity_lock_time', 0);
```

Comment:
Well, the `Do not use Factory in Models` speak for itself. But you are right that it would be a lot of work to add the CoreParametersHelper to every child of FormModel. How about this; lets update the AbstractCommonModel, add the CoreParametersHelper there similar way as the translator is there and then add it to the DI: https://github.com/mautic/mautic/blob/afb4da4491ac2f12e8500d504906bd20f8ff47ee/app/bundles/CoreBundle/DependencyInjection/Compiler/ModelPass.php

This way every model will have access to it.